processing/src/main/java/org/apache/druid/query/rowsandcols/LazilyDecoratedRowsAndColumns.java

```diff
@@ -168,6 +167,11 @@ private void maybeMaterialize()
     }
   }
 
+  private boolean needsMaterialization()
```

Comment:
doubt : why are we materializing eagerly incase of virtual columns (and for that matter ordering as well)?
Also, do we need to do some changes in materialization as well with VCs - I see the following comment in the current code : https://github.com/apache/druid/blob/master/processing/src/main/java/org/apache/druid/query/rowsandcols/LazilyDecoratedRowsAndColumns.java#L336C18-L336C18

Reply:
It's technically theoretically possible to run VirtualColumns on top of the ColumnAccessors instead and do them lazily (i.e. not require materialization to do the VirtualColumns).  But, that requires `ColumnValueSelector` implementations to run, which is just more in-depth work.  The goal of this thing right now is just to get scans to work so that the Drill tests can pass, so I think it's fine to be forcing materialization here.

That said, the comment that you saw makes me think that the scan queries must always be naming the full set of columns (including the virtual columns) which is making this work.  That comment would only apply in the case that the projection columns are not defined (and in that case, I would expect the materialization to not include virtual columns, which is unfortunate)